app/models/manageiq/providers/vmware/infra_manager.rb

```diff
@@ -654,6 +654,43 @@ def find_vm_create_events(vms_list)
       found
     end
 
+    def assign_ems_created_on_queue(vm_ids)
+      MiqQueue.submit_job(
+        :class_name  => self.class.name,
+        :instance_id => id,
+        :method_name => 'assign_ems_created_on',
+        :role        => 'ems_operations',
+        :args        => [vm_ids],
+        :priority    => MiqQueue::MIN_PRIORITY
+      )
+    end
+
+    def assign_ems_created_on(vm_ids)
+      vms_to_update = vms_and_templates.where(:id => vm_ids, :ems_created_on => nil)
+      return if vms_to_update.empty?
+
+      # Of the VMs without a VM create time, filter out the ones for which we
+      #   already have a VM create event
+      vms_to_update = vms_to_update.reject do |v|
+        event = v.ems_events.find_by(:event_type => ["VmCreatedEvent", "VmDeployedEvent"])
```

Comment:
Ugh this is a nasty N+1...I'll put in a follow up PR.

Reply:
Yeah this was a straight move, refactoring to follow

Reply:
Also I should note that this is disabled in Settings by default which is why we probably never noticed when it was in core VmOrTemplate

Reply:
Makes me wonder if we should just get rid of it.

Reply:
Yeah, on vSphere 6.7 we can actually pull this from inventory directly (https://github.com/ManageIQ/manageiq-providers-vmware/pull/645) so this won't be needed at all.

Reply:
Then I'm even _more_ in favor of dropping this.  We can just say that created_on collection is only supported on vsphere 6.7+.

Reply:
:+1: I'm good with removing it then, you want to do that in https://github.com/ManageIQ/manageiq-providers-vmware/pull/645 then?

Reply:
Would like @chessbyte 's input here as well

Reply:
@Fryguy @agrare if this functionality is new to master, then I am ok with it being available on vSphere 6.7+.  If we had this functionality for older vSphere versions, then we need to preserve it, probably by checking for vSphere version and then doing the optimal thing.

Reply:
There is a check at the beginning on the method for whether ems_created_on is already set, so it will naturally skip vsphere 6.7+.  I guess where I'm coming from is that if the setting is `false` by default, then there's likely nobody turning it on.  So, the actual usage of this is probably nobody, hence my thought to drop it.

Reply:
then, I agree to drop it.  if a user complains, we can have a conversation at that time about what to do